src/__builtins.d

```diff
@@ -38,3 +38,23 @@ alias __builtin_va_copy = core.stdc.stdarg.va_copy;
 /* dmd's ImportC rewrites __builtin_va_arg into an instantiation of va_arg
  */
 alias va_arg = core.stdc.stdarg.va_arg;
+
+/*********** floating point *************/
+
+/* https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html
+ */
+
+double __builtin_inf()  { return double.infinity; }
+float  __builtin_inff() { return float.infinity; }
+real   __builtin_infl() { return real.infinity; }
+
+alias __builtin_huge_val  = __builtin_inf;
+alias __builtin_huge_valf = __builtin_inff;
+alias __builtin_huge_vall = __builtin_infl;
+
+import core.stdc.math;
+
+alias __builtin_fabs  = core.stdc.math.fabs;
+alias __builtin_fabsf = core.stdc.math.fabsf;
+alias __builtin_fabsl = core.stdc.math.fabsl;
```

Comment:
version(DigitalMars). @kinke how redundant are these in ldc?

Reply:
AFAICT, LDC needs these too - LDC/LLVM doesn't support any magic `__builtin_*` names, and I haven't found these in generated module `ldc.gccbuiltins_x86` either.

Reply:
It may well do, they don't occur in `ldc.gccbuiltins_x86` because we [filter out]( https://github.com/ldc-developers/ldc/blob/master/utils/gen_gccbuiltins.cpp#L94) things that don't belong to specifically that arch. If LLVM/Clang have these builtins they will be generic. Don't know what "arch" prefix they have though.